libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularPermissionItem.tsx

```diff
@@ -1,38 +1,80 @@
-// (C) 2022 GoodData Corporation
+// (C) 2022-2023 GoodData Corporation
+
+import React, { useCallback } from "react";
+import { useIntl } from "react-intl";
 
-import React from "react";
 import cx from "classnames";
 
-import { GranteeItem, IGranularPermissionTypeItem } from "../types";
+import { GranteeItem, IGranularGrantee, IGranularPermissionTypeItem } from "../types";
+
+import { Bubble, BubbleHoverTrigger } from "../../../../Bubble";
+import { IAlignPoint } from "../../../../typings/positioning";
+
+const alignPoints: IAlignPoint[] = [{ align: "cr cl" }];
 
 interface IGranularPermissionItemProps {
+    grantee: IGranularGrantee;
     permission: IGranularPermissionTypeItem;
+    selectedPermission: string;
     toggleDropdown: () => void;
     onChange: (grantee: GranteeItem) => void;
+    handleSetSelectedPermission: (permission: string) => void;
 }
 
-export const GranularPermissionItem: React.FC<IGranularPermissionItemProps> = ({
+const DropdownItem: React.FC<IGranularPermissionItemProps> = ({
     permission,
+    grantee,
+    selectedPermission,
     toggleDropdown,
+    handleSetSelectedPermission,
+    onChange,
 }) => {
-    const handleOnChange = () => {
-        toggleDropdown();
-        // TODO: change permission
-        // onChange();
-    };
+    const intl = useIntl();
+
+    const handleOnChange = useCallback(
+        (permission: IGranularPermissionTypeItem) => {
+            toggleDropdown();
+            handleSetSelectedPermission(permission.id);
+            onChange({ ...grantee, permissions: [permission.id] });
+        },
+        [permission],
```

Comment:
Missing dependency  toggleDropdown, onChange, toggleDropdown

Reply:
Why there is permissions in dependency that you get it as a parameter ? With one is used now the one from props or the one as function parameter ?

Reply:
This is a great catch, thank you :)